bindings/bindingtester/tests/scripted.py

```diff
@@ -34,10 +34,11 @@
 
 
 class ScriptedTest(Test):
+    MIN_API_VERSION = 500
     TEST_API_VERSION = 520
 
     def __init__(self, subspace):
-        super(ScriptedTest, self).__init__(subspace, ScriptedTest.TEST_API_VERSION, ScriptedTest.TEST_API_VERSION)
+        super(ScriptedTest, self).__init__(subspace, ScriptedTest.MIN_API_VERSION, ScriptedTest.TEST_API_VERSION)
```

Comment:
The purpose of the scripted test was to be a quick predictable test for a binding implementation. I think this somewhat goes against that by having the script generate a random API version by default. I'm not particularly opposed to allowing other API versions (though the scripted test wasn't designed for this in general -- I think it gets ugly to have version-based branches in its spec), but I would prefer to see it use the current API version by default. There's not a mechanism for that currently, though.

The ideal solution to this in my opinion would be to put it in another test. For example, if we need what is essentially a unit test of versionstamps, we could write a new binding test for it that restricted its API version to a particular range that satisfies our criteria. Or better yet, since this bug is not in the bindings at all, we could have a test for it in the client library.

Reply:
Okay, I think there is some discussion to be had about how we can best test API version compatibility across versions. I've reverted this change and added a test to the VersionStampWorkload that will directly test cross-version compatibility of this feature. (I've also verified that that test finds the bug.) So this doesn't have to block this PR.